packages/cardpay-sdk/sdk/prepaid-card/v0.10.0.ts

```diff
@@ -0,0 +1,49 @@
+import Base from './base';
+import BN from 'bn.js';
+import { getAddress } from '../../contracts/addresses';
+import { ZERO_ADDRESS } from '../constants';
+import ERC677ABI from '../../contracts/abi/erc-677';
+import { AbiItem } from 'web3-utils';
+import Web3 from 'web3';
+import { Signer } from 'ethers';
+
+export default class PrepaidCard extends Base {
```

Comment:
generally i think it's better to do this in the other way--make the most recent version the default (so update base with this logic), and then make 0.9.7 the exceptional case. this way it make it easier to continue to evolve the SDK then the old behavior is the exceptional thing.

Reply:
Now that I'm thinking about this, does it even make sense to make an exception for 0.9.0 if we move the new behaviour to the base class?

If we change `getCreateCardPayload` in the *base* class, and extend both 0.9.0 and 0.10.0 from the base without modifications, and deploy the SDK (while not deploying the protocol yet):
- The SDK will send 2 additional params in the data field (2 zero addresses), but the 0.9.0 `PrepaidCardManager` [will just ignore it](https://github.com/cardstack/card-pay-protocol/blob/main/contracts/PrepaidCardManager.sol#L232) because it only grabs the first 5 params in the `data` field - even though `data` has 2 more fields available (2 zero addresses), and the contract will continue executing as it should.

And when we deploy the protocol:
- The SDK behaviour will be the same but this time the 0.10.0 `PrepaidCardManager` will be able to recognize 2 additional params and choose the right execution path (there will be 2 paths in 0.10.0)

The claim where I say that `abi.decode` will just ignore the two additional params is just a hunch which I still need to verify. It's maybe also a bit odd to send params and let the handler not care about it. On the other hand this situation will be temporary and allows us to not make SDK exceptions for 0.9.0. What do you think?

/cc @alex-cardstack

Reply:
But when we try to encode with missing params it will encode to a very different value than not even having the params.

Reply:
Specifically we need to tell the ABI encoder to use a different set of types.

Reply:
You can use this online ABI encoder to see the difference https://adibas03.github.io/online-ethereum-abi-encoder-decoder/#/

Reply:
Ok thanks. I thought if we just append types then it wouldn't make a difference for the rest of the encoded params.